content/master/guides/composition-functions.md

```diff
@@ -0,0 +1,1002 @@
+---
+title: Composition Functions
+state: alpha
+---
+
+Composition Functions allow you to supplement or replace your Compositions with
+advanced logic. You can build a Function using general purpose programming
+languages such as Go or Python, or relevant tools such as Helm, kustomize, or
+CUE. Functions compliment contemporary "Patch and Transform" (P&T) style
+Composition - it's possible to use only P&T, only Functions, or a mix of both in
+the same Composition.
+
+```yaml
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+metadata:
+  name: example
+spec:
+  compositeTypeRef:
+    apiVersion: database.example.org/v1alpha1
+    kind: XPostgreSQLInstance
+  functions:
+  - name: my-cool-Function
+    type: Container
+    container:
+      image: xpkg.upbound.io/my-cool-Function:0.1.0
+```
+
+A Composition Function is a short-lived OCI container that tells Crossplane how
+to reconcile a Composite Resource (XR). The preceding example shows a minimal
+`Composition` that uses a Composition Function. Note that it has a `functions`
+array rather than the typical P&T style array of `resources`.
+
+## Enabling Functions
```

Comment:
I'll update, but I'm curious about the decision. Why use sentence case? I feel like I more commonly see title case.

Reply:
I notice the APA style guide referenced by the docs style guide points out title case as something that should be used for "all headings within a work", but doesn't mention that for sentence case.

* https://apastyle.apa.org/style-grammar-guidelines/capitalization/title-case
* https://apastyle.apa.org/style-grammar-guidelines/capitalization/sentence-case

Reply:
Common documentation styling is sentence case.

https://developers.google.com/style/capitalization#capitalization-in-titles-and-headings
https://learn.microsoft.com/en-us/style-guide/capitalization#sentence-style-capitalization-in-titles-and-headings

Reply:
Thanks. Links like this would be helpful to include in the style guide. I know the _decision_ is captured in the guide, but it helps a lot to be able to read up the rationale behind the decision.

Reply:
The docs will enable [Vale](https://vale.sh/) soon, using the [Upbound Vale library](https://github.com/upbound/vale).

This will put most of these checks as part of CI and provide links to the justification for the rules. This is hinted at in the contrib guide, but not enforced-
https://docs.crossplane.io/v1.10/contributing/docs/#style-guidelines